### PR TITLE
docs: update kubernetes document

### PR DESCRIPTION
**Pull request by liangyawang**
### Description

Add an example of route creation with auto-discovery type Kubernetes

fix https://github.com/apache/apisix/issues/7199

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/zh/latest/discovery/kubernetes.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ discovery:
       port: ${KUBERNETES_SERVICE_PORT}  #default ${KUBERNETES_SERVICE_PORT}
 
     client:
-      # serviceaccount token or token_file
+      # serviceaccount token or token_file token_file中应注意不要有换行符 请使用echo -n token > token_file 来生成此文件
```
Comment (hf400159) [inline docs/zh/latest/discovery/kubernetes.md:48]:
```suggestion
      # serviceaccount token or token_file token_file 中应注意不要有换行符 请使用 echo -n token > token_file 来生成此文件
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 2]
```diff
@@ -147,6 +147,54 @@ nodes("default/plat-dev:3306") 调用会得到如下的返回值：
...
+    Name     Port  Protocol
+    ----     ----  --------
+    8080tcp  8080  TCP                        #注意此处的端口名称 创建路由时 service_name字段中需要添加
+
+Events:  <none>
+```
+
+- 通过apisix api创建自动发现类型为kubernetes的路由 示例如下：
+```
+curl http://127.0.0.1:80/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f8sdfcx7ad84b625c8f1sdfagw32' -X PUT -i -d '
```
Comment (hf400159) [inline docs/zh/latest/discovery/kubernetes.md:179]:
```suggestion
curl http://127.0.0.1:80/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -i -d '
```

Comment (hf400159) [reply docs/zh/latest/discovery/kubernetes.md:179]: Please use the default Admin API Key.

**Review by hf400159 — COMMENTED** [review]
Can you modify the format of this document? please refer: [Chinese documentation style guide](https://apisix.apache.org/zh/docs/general/documentation-style-guide).

**Comment by liangyawang** [comment]
这个邮件很乱 您是让我修改哪部分的格式 咱们可否在slack交流

**Comment by hf400159** [comment]
> hi, I have sent you a private message. 😄

File: docs/zh/latest/discovery/kubernetes.md (hunk 1 above)
Comment (tokers) [reply docs/zh/latest/discovery/kubernetes.md:48]: It's better to show the shell command in a code block.

```shell
echo -n token > token_file
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 3]
```diff
@@ -147,6 +147,54 @@ nodes("default/plat-dev:3306") 调用会得到如下的返回值：
        },
    }
   ```
+## kubernetes类型的自动发现 路由创建示例
```
Comment (tokers) [inline docs/zh/latest/discovery/kubernetes.md:150]:
```suggestion

## kubernetes 类型的自动发现 路由创建示例
```

File: docs/zh/latest/discovery/kubernetes.md [hunk 4]
```diff
@@ -147,6 +147,54 @@ nodes("default/plat-dev:3306") 调用会得到如下的返回值：
...
+
+- 服务、对应名称空间及endpoints信息展示
+```
+[root@ ~]# kubectl get pods -n test |grep test-app
```
Comment (tokers) [inline docs/zh/latest/discovery/kubernetes.md:154]: We may remove these leaded prompts. Just reserve the command itself.

**Comment by Maple521** [comment]
I have met the same problem, but I still can't solve it according to the above plan.

image
`
discovery:
  kubernetes:
    service:
      schema: https
      host: "192.168.65.4"
      port: "6443"
    client:
      token: |-  
        eyJhbGciOixxxxxxxx
      #token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
    namespace_selector:
      equal: service-k8s-demo`

my upstream service config is：
`{
  "timeout": {
    "connect": 6,
    "send": 6,
    "read": 6
  },
  "type": "roundrobin",
  "scheme": "http",
  "discovery_type": "kubernetes",
  "pass_host": "pass",
  "name": "get service-provider service from K8s",
  "service_name": "service-k8s-demo/service-provider:tcp30000",
  "keepalive_pool": {
    "idle_timeout": 60,
    "requests": 1000,
    "size": 320
  }
}`

Then when I ask the Apisix gateway, it will report an error：
`172.19.0.1 - - [10/Mar/2023:06:38:37 +0000] hongchenkezhan.com "GET /provider/provider-hello HTTP/1.1" 503 269 0.000 "-" "PostmanRuntime-ApipostRuntime/1.1.0" - - - "http://hongchenkezhan.com"
2023/03/10 06:38:47 [error] 50#50: *53318 [lua] init.lua:486: handle_upstream(): failed to set upstream: no valid upstream node: nil, client: 172.19.0.1, server: _, request: "GET /provider/provider-hello HTTP/1.1", host: "hongchenkezhan.com"`

image

I obtained the token through this command：
`kubectl -n service-k8s-demo  exec -it service-provider-6d47bdd64-4zqhd -- cat /var/run/secrets/kubernetes.io/serviceaccount/token`

This is the configuration of my serviceaccount. It has permissions：
`apiVersion: v1
kind: Namespace
metadata:
  name: service-k8s-demo
  labels:
    name: service-k8s-demo

---

apiVersion: v1
kind: ServiceAccount
metadata:
  name: service-k8s-demo
  namespace: service-k8s-demo

---

kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  namespace: service-k8s-demo
  name: service-k8s-demo
rules:
  - apiGroups:
    - ""
    resources:
    - services
    - configmaps
    - endpoints
    - nodes
    - pods
    - secrets
    - namespaces
    verbs:
    - get
    - list
    - watch

---

apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: service-k8s-demo
  namespace: service-k8s-demo
subjects:
- kind: ServiceAccount
  name: service-k8s-demo
  namespace: service-k8s-demo
roleRef:
  kind: ClusterRole
  name: service-k8s-demo
  apiGroup: rbac.authorization.k8s.io`

**Pull request closed without merging**